code/obj/flock/structure/fabricator.dm

```diff
@@ -0,0 +1,77 @@
+	health = 20
+	health_max = 20
+	repair_per_resource = 1
+	passthrough = TRUE
+
+	var/resources_to_produce = 0
+
+	New(atom/location, datum/flock/F = null, obj/content_holder)
+		..()
+		if (istype(content_holder, /obj/machinery/vending))
+			var/obj/machinery/vending/vending_machine = content_holder
+			for (var/datum/data/vending_product/product as anything in vending_machine.product_list)
+				src.resources_to_produce += get_initial_item_health(product.product_path) * product.product_amount
+		else if (istype(content_holder, /obj/machinery/manufacturer))
+			var/obj/machinery/manufacturer/fab = content_holder
+			if (fab.free_resource_amt)
+				for (var/obj/item/material_piece/mat as anything in fab.free_resources)
+					src.resources_to_produce += get_initial_item_health(mat) * fab.free_resource_amt
+			else
+				for (var/obj/item/material_piece/mat in fab.contents)
+					src.resources_to_produce += get_initial_item_health(mat) * mat.amount
+			src.resources_to_produce += round(fab.reagents.total_volume / 40)
+		else if (istype(content_holder, /obj/submachine/seed_vendor))
+			src.resources_to_produce += 100
+		else if (istype(content_holder, /obj/machinery/dispenser))
+			var/obj/machinery/dispenser/tank_holder = content_holder
+			src.resources_to_produce += tank_holder.o2tanks * get_initial_item_health(/obj/item/tank/oxygen) + tank_holder.pltanks * get_initial_item_health(/obj/item/tank/plasma)
+		else if (istype(content_holder, /obj/machinery/disposal_pipedispenser))
+			src.resources_to_produce += 100
+		else if (istype(content_holder, /obj/machinery/chem_dispenser) || istype(content_holder, /obj/machinery/chemicompiler_stationary))
+			src.resources_to_produce += 100
+		else if (istype(content_holder, /obj/reagent_dispensers))
+			var/obj/reagent_dispensers/reagent_holder = content_holder
+			src.resources_to_produce += round(reagent_holder.reagents.total_volume / 40) // welding fuel dispenser has 4000u of fuel, for 100 resources
+
+		src.info_tag.set_info_tag("Resources left: [src.resources_to_produce]")
+
+		if (!src.resources_to_produce)
+			SPAWN(0.1 SECONDS)
+				if (src)
+					flock_speak(src, "ALERT: No resources available to produce.", src.flock)
+		else
+			ON_COOLDOWN(src, "resource_production", 10 SECONDS)
+
+	building_specific_info()
+		return "<span class='bold'>Resources left to produce:</span> [src.resources_to_produce]."
+
+	process(mult)
+		if (!src.resources_to_produce)
+			return
+		if (ON_COOLDOWN(src, "resource_production", 10 SECONDS))
+			return
+
+		var/obj/item/flockcache/resource_cube = new(get_turf(src))
+		resource_cube.resources = min(src.resources_to_produce, round(25 * mult))
+		playsound(src, 'sound/effects/crackle3.ogg', 40, TRUE, -10) // placeholder
+		src.resources_to_produce -= resource_cube.resources
+
+		src.info_tag.set_info_tag("Resources left: [src.resources_to_produce]")
+
+		if (!src.resources_to_produce)
+			flock_speak(src, "ALERT: No resources left to produce", src.flock)
```

Comment:
Would make more sense to gib on completion I think, to make it clearer to the flockmind that this building is over and done with. Right now it's left in an animated state and looks like it should still be doing something.